dali/imgcodec/image_format.cc

```diff
@@ -67,5 +67,19 @@ span<ImageFormat* const> ImageFormatRegistry::Formats() const {
   return make_cspan(format_ptrs_);
 }
 
+size_t ImageParser::ReadHeader(ImageSource *encoded, uint8_t *buffer, size_t n) const {
```

Comment:
We use memcpy order (output goes first)
```suggestion
size_t ImageParser::ReadHeader(uint8_t *buffer, ImageSource *encoded, size_t n) const {
```